lib/srv/server/ec2_watcher.go

```diff
@@ -81,6 +90,34 @@ func ToEC2Instances(insts []*ec2.Instance) []EC2Instance {
 
 }
 
+// ServerInfos creates a ServerInfo resource for each discovered instance.
+func (i *EC2Instances) ServerInfos() ([]types.ServerInfo, error) {
```

Comment:
Suggestion: Moving the "covert" logic to `services/server_info.go`. This makes it easier to keep consistency across different providers (if/when added) and better for testing (which might be necessary to cover here too). We follow this when creating databases from the AWS SDK structs ([here is an example](https://github.com/gravitational/teleport/blob/master/lib/services/database.go#L693) and [its tests](https://github.com/gravitational/teleport/blob/master/lib/services/database_test.go#L710-L731)).

Reply:
Definitely on board, but doing this right now leads to an import cycle between `lib/services` and `lib/labels`.